qemu/tests/qemu_guest_agent.py

```diff
@@ -390,6 +393,269 @@ def gagent_check_set_vcpus(self, test, params, env):
+        Execute "guest-get-time" command to guest agent
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        timeout = float(params.get("login_timeout", 240))
+        session = self.vm.wait_for_login(timeout=timeout)
+        os_type = params["os_type"]
+        error.context("get the time of the guest", logging.info)
+        nanoseconds_time = self.gagent.get_time()
+        if nanoseconds_time is False:
+            raise error.TestNAError("the  guest-get-time cmd is not supported")
+        error.context("the time get by guest-get-time is '%d' "
+                      % nanoseconds_time, logging.info)
+        guest_time = utils_misc.get_guest_os_datetime(session, os_type)
+        if not guest_time:
+            raise error.TestError("can't get the guest time for contrast")
+        error.context("the time get inside guest by shell cmd is '%d' "
+                      % guest_time, logging.info)
+        delta = abs(guest_time - nanoseconds_time / 1000000000)
+        if delta > 3:
+            raise error.TestFail("the time get by guest agent is not the same "
+                                 "with that by time check cmd inside guest")
+
+    @error.context_aware
+    def gagent_check_set_time(self, test, params, env):
+        """
+        Execute "guest-set-time" command to guest agent
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        timeout = float(params.get("login_timeout", 240))
+        session = self.vm.wait_for_login(timeout=timeout)
+        os_type = params["os_type"]
+        error.context("get the time of the guest", logging.info)
+        guest_time_before = utils_misc.get_guest_os_datetime(session, os_type)
+        if not guest_time_before:
+            raise error.TestError("can't get the guest time for contrast")
+        error.context("the time before being moved back into past  is '%d' "
+                      % guest_time_before, logging.info)
+        # Need to move the guest time one week into the past
+        target_time = (guest_time_before - 604800) * 1000000000
+        ret = self.gagent.set_time(target_time)
+        if ret is False:
+            raise error.TestNAError("the guest-set-time cmd is not supported")
+        guest_time_after = utils_misc.get_guest_os_datetime(session, os_type)
+        error.context("the time after being moved back into past  is '%d' "
+                      % guest_time_after, logging.info)
+        delta = abs(guest_time_after - target_time / 1000000000)
+        if delta > 3:
+            raise error.TestFail("the time set for guest is not the same "
+                                 "with target")
+        # Set the system time from the hwclock
+        move_time_cmd = params["move_time_cmd"]
+        session.cmd("hwclock -w")
```

Comment:
Windows test failed, is it linux only scenario?
15:46:33 ERROR| ShellCmdError: Shell command failed: 'hwclock -w'    (status: 9009,    output: "'hwclock' is not recognized as an internal or external command,\noperable program or batch file.\n\n")